compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -668,8 +668,21 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
         let mut let_suggestion = None;
         let mut misc_suggestion = None;
         let mut interpreted_as_const = None;
+        let mut interpreted_as_const_sugg = None;
 
-        if let PatKind::Constant { .. }
+        if let PatKind::Constant { opt_def: Some(def_id), .. }
+        | PatKind::AscribeUserType {
+            subpattern: box Pat { kind: PatKind::Constant { opt_def: Some(def_id), .. }, .. },
+            ..
+        } = pat.kind
+            && let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(pat.span)
```

Comment:
Ah, you read my mind with the `DefId` -- but we can take it further and use the `tcx.item_name` here to get the name of the constant. This will regress reexported constants, but is more resilient IMO and is worth the overhead.

Reply:
Specifically, this pat span will not possibly be tied to arbitrary source code, in `pat_macro!(/* uwu */)`.